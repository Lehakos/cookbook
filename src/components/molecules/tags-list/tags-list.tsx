import React, { ReactNode } from 'react';

import { Tag } from 'components/atoms';

import * as s from './tags-list.styles';

interface Item {
  id: string;
  value: ReactNode;
  color?: string;
}

interface Props {
  items: Item[];
}

export const TagsList: React.FunctionComponent<Props> = function({ items }) {
  return (
    <s.Wrapper>
      {items.map(item => (
        <s.Item key={item.id}>
          <Tag color={item.color}>{item.value}</Tag>
        </s.Item>
      ))}
    </s.Wrapper>
  );
};
