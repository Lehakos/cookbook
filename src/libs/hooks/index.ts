export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
