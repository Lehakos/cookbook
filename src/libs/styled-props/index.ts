export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
