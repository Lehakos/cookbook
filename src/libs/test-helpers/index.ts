export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
