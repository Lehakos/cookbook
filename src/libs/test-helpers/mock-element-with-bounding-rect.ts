export const mockElementWithBoundingRect = ({
  width = 100,
  height = 100,
  top = 0,
  left = 0,
}) => {
  const right = left + width;
  const bottom = top + height;

  Object.defineProperties(Element.prototype, {
    offsetLeft: {
      get() {
        return left;
      },
      configurable: true,
    },
    offsetTop: {
      get() {
        return top;
      },
      configurable: true,
    },
    offsetHeight: {
      get() {
        return height;
      },
      configurable: true,
    },
    offsetWidth: {
      get() {
        return width;
      },
      configurable: true,
    },
  });

  Element.prototype.getBoundingClientRect = jest.fn(() => {
    return {
      width,
      height,
      top,
      left,
      bottom,
      right,
    };
  });

  const element = document.createElement('div');

  return element;
};
