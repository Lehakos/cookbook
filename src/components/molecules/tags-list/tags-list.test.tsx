import React from 'react';

import { render } from 'libs/test-helpers';

import { TagsList } from './tags-list';

describe('<TagsList />', () => {
  it('renders properly', () => {
    const { container } = render(<TagsList items={[]} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
