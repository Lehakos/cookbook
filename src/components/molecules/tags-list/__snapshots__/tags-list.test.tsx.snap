// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagsList /> renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: -8px;
  padding: 0;
  min-width: 100%;
  list-style: none;
}

<ul
  class="c0"
/>
`;
